### DISCUSSION TITLE
Popover problems in ios 17

### DISCUSSION THREAD

**Discussion by jsonengler** [appcelerator/titanium_mobile#13912] [Q&A] [answered]
We've got 3 ipad applications. we use popovers a lot, and it seems like they all have problems with ios 17. The popovers show up, but the alignment is all messed up so things won't show up right. When we force rotate the app, then it fixes itself. It works fine in ios 16.

Anyone have suggestions on how to fix this? I've been searching for popover but i don't see anyone else report it. i am using 12.2.0.GA

**Comment by de1mat** [comment]
Do you have a screenshot to show the problem @jsonengler ?

**Comment by jsonengler** [comment]
@de1mat 

oops! i forgot the image, sorry about that

<img width="443" alt="FM-11367-States-crooked" src="https://github.com/tidev/titanium-sdk/assets/15806921/82e5fd90-6714-40b9-8173-b6854fe31e85">

Here's an example where we have a popover with a table. it's getting cut off a bit on the left side. when rotating, it fixes it
```
<Alloy>
    <Popover id='popover' class='popover' ns="Ti.UI.iPad">
        <ContentView>
            <Window>
                <TableView id='table' onClick='onTableClick'/>
            </Window>
        </ContentView>
    </Popover>
</Alloy>
```


tss:
```
".popover" : {
    height: Ti.UI.SIZE,
    width: Ti.UI.SIZE,
    arrowDirection:Ti.UI.iPad.POPOVER_ARROW_DIRECTION_RIGHT
}
"#table" : {
    height: Ti.UI.SIZE,
    width: 200
}
```

we reuse this one a lot, but we have a few others that also have problems

edit: looks like it's getting shifted in the direction of the arrow. most of our popovers have an arrow going left and they shift left. we have one where the arrow points down, and it's shifted down.

**Reply by m1ga** [reply]
just to link this here too: https://github.com/tidev/titanium-sdk/pull/13916
Possible fix made by Hans. Please test it if you can

**Comment by AhmedMSayed** [comment]
have the same issue, my app depands on popovers a lot

**Comment by m1ga** [comment, marked as answer]
[Fix](https://github.com/tidev/titanium-sdk/pull/13929) is included in 12.2.1. You can grab the build from https://downloads.titaniumsdk.com/builds/12_2_X or wait for the release next week
